File: nandincpp/Destroying Asteroids.cpp
class Solution {
public:
    bool asteroidsDestroyed(int mass, vector<int>& a)
    {
        bool flag = 1;
        sort(a.begin(), a.end());
        long long m = mass;
            
        for(int i=0;i<a.size();i++)
        {
            if(m >= a[i])
            {
                m += (long long)a[i];
            }
            else
            {
                flag = 0;
                break;
            }
        }
        
        return flag;
    }
};

File: nandincpp/Destroying Asteroids.java
class Solution {
    public boolean asteroidsDestroyed(int mass, int[] a) 
    {
        Arrays.sort(a);
        long m = mass;
        boolean flag = true;
        
        for(int i=0;i<a.length;i++)
        {
            if(m >= a[i])
                m += a[i];
            else
            {
                flag = false;
                break;
            }
        }
        
        return flag;
    }
}